.gitignore

```diff
@@ -11,6 +11,7 @@ installed
 /composer.phar
 _ide_helper.php
 *.swp
+composer.lock
```

Comment:
Why add composer.lock to gitignore? I like to keep it in git, so that there is always a lit of dependencies that are known to work. You can call `composer update` on your own fork/install whenever you want.
https://getcomposer.org/doc/01-basic-usage.md#commit-your-composer-lock-file-to-version-control

Reply:
I feel this needs to be a separate PR as to discuss the reasons why we need to ignore or not ignore this file. Please undo this change from this PR for the time being? 🙏